Makefile.standalone

```diff
@@ -0,0 +1,44 @@
+
+## path configuration (FHS-style)
+PREFIX  ?= /usr
+SBINDIR ?= $(PREFIX)/sbin
+
+## toolchain commands
+export GO ?= go
+export LDFLAGS=-s -w
+
+PROGS=$(CURDIR)/dockerd $(CURDIR)/docker-proxy
+
+all: compile
+
+compile: $(PROGS)
+
+## generate go.mod from vendor tree
+$(CURDIR)/go.mod: vendor/modules.txt
+	@( \
+		echo "module github.com/docker/docker" ; \
+		echo "" ; \
+		echo "go 1.19" ; \
+		echo "" ; \
+		echo "require (" ; \
+		cat $< | grep -E '^# ' | grep -Ev '^## ' | sed -e 's~^# ~\t~;' ; \
+		echo ")" ; \
+	) > $@
```

Comment:
There should be no need to generate the module; you can use the `vendor.mod` and `vendor.sum`, which are `go.mod` files, but with a different name.

If needed you can symlink them to `go.mod` during these steps, or create a "stub" `go.mod` and use the `-modfile=vendor.mod` option (see https://github.com/moby/moby/blob/6b2ee07a263a32535bb8dfcb732313b79c4dcdc1/hack/with-go-mod.sh)